src/connectors/onchain_events/mod.rs

```diff
@@ -0,0 +1,361 @@
+                let fid = Uint::to::<u64>(&id);
+                add_event(
+                    fid,
+                    EventType::IdRegister(IdRegisterEvent::Transfer { to, from }),
+                );
+                return Ok(());
+            }
+            Some(&IdRegistryAbi::ChangeRecoveryAddress::SIGNATURE_HASH) => {
+                let IdRegistryAbi::ChangeRecoveryAddress { id, recovery } =
+                    event.log_decode()?.inner.data;
+                let fid = Uint::to::<u64>(&id);
+                add_event(
+                    fid,
+                    EventType::IdRegister(IdRegisterEvent::ChangeRecovery {
+                        recovery_address: recovery,
+                    }),
+                );
+                return Ok(());
+            }
+            Some(&KeyRegistryAbi::Add::SIGNATURE_HASH) => {
+                let KeyRegistryAbi::Add {
+                    fid,
+                    key: _,
+                    keytype,
+                    keyBytes,
+                    metadatatype,
+                    metadata,
+                } = event.log_decode()?.inner.data;
+                let fid = Uint::to::<u64>(&fid);
+                add_event(
+                    fid,
+                    EventType::Signer(SignerEvent::Add {
+                        key: keyBytes,
+                        key_type: keytype,
+                        metadata,
+                        metadata_type: metadatatype,
+                    }),
+                );
+                return Ok(());
+            }
+            Some(&KeyRegistryAbi::Remove::SIGNATURE_HASH) => {
+                let KeyRegistryAbi::Remove {
+                    fid,
+                    key: _,
+                    keyBytes,
+                } = event.log_decode()?.inner.data;
+                let fid = Uint::to::<u64>(&fid);
+                add_event(
+                    fid,
+                    EventType::Signer(SignerEvent::Remove { key: keyBytes }),
+                );
+                return Ok(());
+            }
+            Some(&KeyRegistryAbi::AdminReset::SIGNATURE_HASH) => {
+                let KeyRegistryAbi::AdminReset {
+                    fid,
+                    key: _,
+                    keyBytes,
+                } = event.log_decode()?.inner.data;
+                let fid = Uint::to::<u64>(&fid);
+                add_event(
+                    fid,
+                    EventType::Signer(SignerEvent::AdminReset { key: keyBytes }),
+                );
+                return Ok(());
+            }
+            Some(&KeyRegistryAbi::Migrated::SIGNATURE_HASH) => {
+                let KeyRegistryAbi::Migrated { keysMigratedAt } = event.log_decode()?.inner.data;
+                let migrated_at = Uint::to::<u64>(&keysMigratedAt);
+                add_event(0, EventType::SignerMigrated { migrated_at });
+                return Ok(());
+            }
+            _ => return Ok(()),
```

Comment:
What events might captured by this path? Is it possible to cover all cases? (my guess is no, but not sure).

Probably worth logging/warning something here? Or would it be too noisy?

Reply:
There are a bunch of events we don't process. I'm not sure how often they actually show up, though. IMO it seems ok to just leave this codepath without logging--- we don't actually need to know if there are other events being emitted.